tests/engine/adhoc_test.go

```diff
@@ -241,41 +239,41 @@ func importedAndExportedFunc(t *testing.T, newEngine func() *wazero.Engine) {
 	require.NoError(t, err)
 
 	var memory *wasm.MemoryInstance
-	storeInt := func(ctx publicwasm.HostFunctionCallContext, offset uint32, val uint64) uint32 {
+	storeInt := func(ctx publicwasm.ModuleContext, offset uint32, val uint64) uint32 {
 		if !ctx.Memory().WriteUint64Le(offset, val) {
 			return 1
 		}
-		// sneak a reference to the memory so we can check it later
+		// sneak a reference to the memory, so we can check it later
 		memory = ctx.Memory().(*wasm.MemoryInstance)
 		return 0
 	}
 
-	hostFuncs, err := wazero.NewHostFunctions(map[string]interface{}{"store_int": storeInt})
-	require.NoError(t, err)
+	store := wazero.NewStoreWithConfig(&wazero.StoreConfig{Engine: newEngine()})
 
-	store, err := wazero.NewStoreWithConfig(&wazero.StoreConfig{
-		Engine:                newEngine(),
-		ModuleToHostFunctions: map[string]*wazero.HostFunctions{"": hostFuncs},
-	})
+	_, err = wazero.ExportHostFunctions(store, "", map[string]interface{}{"store_int": storeInt})
 	require.NoError(t, err)
 
-	m, err := store.Instantiate(mod)
+	exports, err := wazero.InstantiateModule(store, mod)
 	require.NoError(t, err)
 
 	// Call store_int and ensure it didn't return an error code.
-	storeIntFn, ok := m.GetFunctionI32Return("store_int")
+	storeIntFn, ok := exports.Function("store_int")
 	require.True(t, ok)
 
-	result, err := storeIntFn(ctx, 1, math.MaxUint64)
+	results, err := storeIntFn(context.Background(), 1, math.MaxUint64)
 	require.NoError(t, err)
-	require.Equal(t, uint32(0), result)
+	require.Equal(t, uint64(0), results[0])
 
 	// Since offset=1 and val=math.MaxUint64, we expect to have written exactly 8 bytes, with all bits set, at index 1.
 	require.Equal(t, []byte{0x0, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x0}, memory.Buffer[0:10])
 }
 
-//  hostFunctions ensures arg0 is optionally a context, and fails if a float parameter corrupts a host function value
-func hostFunctions(t *testing.T, newEngine func() *wazero.Engine) {
+func TestHostFunctions(t *testing.T) {
+	testHostFunctions(t, wazero.NewEngineInterpreter)
```

Comment:
lets's run this for JIT as well :D

Reply:
accident.. the way we run tests makes it hard to run one. Later I will refactor but for now deleted this as it is already called